win32/src/threadpool_win32.c

```diff
@@ -583,36 +605,44 @@ int threadpool_timer_start(THANDLE(THREADPOOL) threadpool, uint32_t start_delay_
                 }
                 else
                 {
-                    timer_temp->timer = tp_timer;
-                    timer_temp->work_function = work_function;
-                    timer_temp->work_function_context = work_function_context;
-
-                    /* Codes_SRS_THREADPOOL_WIN32_42_007: [ threadpool_timer_start shall call SetThreadpoolTimer, passing negative start_delay_ms as pftDueTime, timer_period_ms as msPeriod, and 0 as msWindowLength. ]*/
-                    threadpool_internal_set_timer(tp_timer, start_delay_ms, timer_period_ms);
-
-                    /* Codes_SRS_THREADPOOL_WIN32_42_009: [ threadpool_timer_start shall return the allocated handle in timer_handle. ]*/
-                    *timer_handle = timer_temp;
-                    timer_temp = NULL;
-
-                    /* Codes_SRS_THREADPOOL_WIN32_42_010: [ threadpool_timer_start shall succeed and return 0. ]*/
-                    result = 0;
+                    timer_ptr->timer = tp_timer;
+
+                    /* Codes_SRS_THREADPOOL_WIN32_07_002: [ threadpool_timer_start shall initialize the a lock by calling srw_lock_ll_init. ]*/
+                    if (srw_lock_ll_init(&timer_ptr->timer_lock) != 0)
+                    {
+                        /* Codes_SRS_THREADPOOL_WIN32_42_008: [ If any error occurs, threadpool_timer_start shall fail and return a non-zero value. ]*/
+                        LogError("srw_lock_ll_init failed");
+                        result = MU_FAILURE;
+                    }
+                    else
+                    {
+                        timer_ptr->work_function = work_function;
+                        timer_ptr->work_function_context = work_function_context;
+
+                        /* Codes_SRS_THREADPOOL_WIN32_42_007: [ threadpool_timer_start shall call SetThreadpoolTimer, passing negative start_delay_ms as pftDueTime, timer_period_ms as msPeriod, and 0 as msWindowLength. ]*/
+                        threadpool_internal_set_timer(tp_timer, start_delay_ms, timer_period_ms);
+
+                        /* Codes_SRS_THREADPOOL_WIN32_42_009: [ threadpool_timer_start shall return the allocated handle in timer_handle. ]*/
+                        THANDLE_MOVE(TIMER_INSTANCE)(timer_handle, &timer_temp);

```

Comment:
>THANDLE_MOVE(TIMER_INSTANCE)(timer_handle, &timer_temp); [](http://example.com/codeflow?start=24&length=56)

Would THANDLE_INITIALIZE_MOVE make more sense here?

Reply:
Same for Linux?

Reply:
done